request.js

```diff
@@ -921,7 +921,7 @@ Request.prototype.onRequestResponse = function (response) {
     })
 
     var responseContent
-    if (self.gzip) {
+    if (self.gzip && self.method !== 'HEAD') {
```

Comment:
This will fix the particular issue with HEAD requests, but will still cause issues if the response is empty for say a GET. A better way of handling would be to catch the Error from zlib.